search.go

```diff
@@ -1,175 +1,152 @@
 package opensearchtools
 
 import (
-	"bytes"
 	"context"
 	"encoding/json"
 	"net/http"
-
-	"github.com/opensearch-project/opensearch-go/v2"
-	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
-
-	"github.com/CrowdStrike/opensearchtools/search"
 )
 
-// SearchRequest wraps the functionality of [opensearchapi.SearchRequest] by supporting request body creation.
-// An empty SearchRequest defaults to a size of 0. While this will find matches and return a total hits value,
-// it will return no documents. It is recommended to use NewSearchRequest or use SetSize.
-// A simple term query search as an example:
+// Search defines a method which knows how to make an OpenSearch [Search] request.
+// It should be implemented by a version-specific executor.
+//
+// [Search]: https://openorg/docs/latest/api-reference/search/
+type Search interface {
+	Search(ctx context.Context, req *SearchRequest) (*SearchResponse, error)
+}
+
+// SearchRequest is a domain model union type for all the fields of a Search request across
+// all supported OpenSearch version.
+// Currently supported versions are:
+//   - OpenSearch 2
+//
+// This SearchRequest is intended to be used along with a version-specific executor such
+// as [opensearchtools.osv2.Executor]. For example:
```

Comment:
Use `/` for the package separator.  Should be `[opensearchtools/osv2.Executor]`.